network-api/networkapi/wagtailpages/pagemodels/mixin/foundation_banner_inheritance.py

```diff
@@ -0,0 +1,20 @@
+# A mixin that imparts banner inheritance for whatever model it gets tacked onto
+class FoundationBannerInheritanceMixin():
+
+    def get_banner(self):
+        if self.banner is None:
+            ancestors = self.get_ancestors()
+            root = ancestors[1].specific
+            # note: we need to reverse get_ancestors, because by default
+            # this gives a list that starts at the root, and then gets more
+            # specific, ending in the direct parent. We want to check in the
+            # opposite direction: start at the parent until we hit the root.
+            for page in ancestors.reverse():
+                page = page.specific
+                if page == root:
+                    return None
+                valid = isinstance(page, FoundationBannerInheritanceMixin)
```

Comment:
Are mixins considered as "class" as well?

I checked the [doc](https://python-reference.readthedocs.io/en/latest/docs/functions/isinstance.html) of `isinstance` and it says "it returns a Boolean stating whether the object is an instance or subclass of another object."

Reply:
They are. Python allows multiple inheritance, so in `class X(A,B,C)` is a class that extends three difference classes, and `isinstance(X,X)`, `isinstance(X,A)`, `isinstance(X,B)`, and `isinstance(X.C)` are all true.

Reply:
Ah gotcha. Didn't scroll down far enough to see this line in another file: 
```class PrimaryPage(FoundationMetadataPageMixin, FoundationBannerInheritanceMixin, Page)```